lib/uiautomator2.js

```diff
@@ -305,6 +305,26 @@ class UiAutomator2Server {
   async cleanupAutomationLeftovers (strictCleanup = false) {
     this.log.debug(`Performing ${strictCleanup ? 'strict' : 'shallow'} cleanup of automation leftovers`);
 
+    const waitStop = async () => {
+      // Wait for the process stop by sending a status request to the port.
+      // We observed the process stop could be delayed, then it caused the process stop
+      // in the middle of session preparation. It caused invalid session error response by the uia2 server,
+      // but that was because the process stop's delay.
+      const timeout = 3000;
+      try {
+        await waitForCondition(async () => {
+          try {
+            await this.jwproxy.command('/status', 'GET');
```

Comment:
I assume jwproxy might not exist on this stage
Either perform requests using raw axios like we do it above or simply check the process existence using processExists API.

Reply:
let me use axios one. thought the process check, but potentially the weird os situation could tell a lie (I don't have the env anymore, so cannot check that though)

Reply:
Updated.

Latest error case:

```
[debug] [AndroidUiautomator2Driver@965b (79f6d2c7)] Performing shallow cleanup of automation leftovers
[debug] [AndroidUiautomator2Driver@965b (79f6d2c7)] No obsolete sessions have been detected (socket hang up)
[debug] [ADB] Running '/Users/kazu/Library/Android/sdk/platform-tools/adb -P 5037 -s D0AA002182JC0202126 shell am force-stop io.appium.uiautomator2.server.test'
[AndroidUiautomator2Driver@965b (79f6d2c7)] The instrumentation process might fail to stop within 3000ms timeout.
[debug] [AndroidUiautomator2Driver@965b (79f6d2c7)] Deleting UiAutomator2 session
[debug] [AndroidUiautomator2Driver@965b (79f6d2c7)] Deleting UiAutomator2 server session
[debug] [AndroidUiautomator2Driver@965b (79f6d2c7)] Matched '/' to command name 'deleteSession'
[AndroidUiautomator2Driver@965b (79f6d2c7)] Did not get confirmation UiAutomator2 deleteSession worked; Error was: UnknownError: An unknown server-side error occurred while processing the command. Original error: Trying to proxy a session command without session id
[debug] [ADB] Running '/Users/kazu/Library/Android/sdk/platform-tools/adb -P 5037 -s D0AA002182JC0202126 shell dumpsys activity services io.appium.settings/.recorder.RecorderService'
[debug] [Logcat] Stopping logcat capture
[debug] [ADB] Removing forwarded port socket connection: 8800
```

Reply:
maybe something else is needed in addition to `force-stop io.appium.uiautomator2.server.test` to fully stop the automation. I expect to not see `[AndroidUiautomator2Driver@965b (79f6d2c7)] The instrumentation process might fail to stop within 3000ms timeout.`

Reply:
Perhaps we should just kill all active instrumentation processes using killProcessesByName ? 
It is just necessary to check the actual ps output while it is running

Reply:
Hm... killing all instrumentation process seems too aggressive...?

What I saw was kind of "delay", but it did not have multiple instrumentation processes. so for now, maybe we can see how this will help...?

Reply:
after thinking a few, probably "waiting for the process might be stopped (no response for the port)" thing is sufficient for my observation at this time.
The session failure would occur in the `createSession`, so existing teardown can cover cases. The purpose of this is to make sure the process stop (and if not, let users know possible issue as a warning), thus just printing log is enough